python/hail/matrixtable.py

```diff
@@ -57,7 +57,7 @@ def _process_keys(self, left):
                     f = self._parent._fields_inverse[v]
                 else:
                     f = Env.get_uid()
-                new_col_fields[f] = v
+                    new_col_fields[f] = v
```

Comment:
I hit problems in the tests here. We don't need to annotate a field if it already exists.

Reply:
actually this fixes  #3721 too